spring-cloud-config-client/src/main/java/org/springframework/cloud/config/client/ConfigServerInstanceProvider.java

```diff
@@ -70,6 +88,10 @@ public interface Function {
 
 		List<ServiceInstance> apply(String serviceId);
 
+		default List<ServiceInstance> apply(String serviceId, Binder binder, BindHandler bindHandler, Log log) {
```

Comment:
I'm a big confused by this. Why have this and the ones passed into the constructor?

Reply:
The ones in the constructor of `ConfigServerInstanceProvider` are then passed when it calls the `Function` above.

Reply:
Why can't constructor binder etc work in all cases?

Reply:
They are the same.

They are passed in the constructor of `ConfigServerInstanceProvider` then used in `ConfigServerInstanceProvider.getConfigServerInstances` when it calls `Function.apply`.

Reply:
Right, the other impls of function need them, not config client